### PR TITLE
Batch with timestamp

### PR DESCRIPTION
**Pull request by xrigau**
Adds a new column that keeps the timestamp of the last modification made to a batch, that is when the status of a batch was last modified.

We also renamed the LAST_MODIFIED column from Downloads so that we can use the same name as the one in Batches.

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/142/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java [hunk 1]
```diff
@@ -39,16 +39,19 @@
...
         this.resolver = resolver;
         this.downloadDeleter = downloadDeleter;
         this.downloadsUriProvider = downloadsUriProvider;
+        this.systemFacade = systemFacade;
     }
 
     void updateBatchStatus(long batchId, int status) {
         ContentValues values = new ContentValues();
         values.put(DownloadContract.Batches.COLUMN_STATUS, status);
+        values.put(DownloadContract.Batches.COLUMN_LAST_MODIFICATION, systemFacade.currentTimeMillis());
```
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: this timestamp is only set when updating the status and not during a download?

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: yep, PR desr. says it
>Adds a new column that keeps the timestamp of the last modification made to a batch, that is when the status of a batch was last modified.

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: as in update of the last modified column during a download is in the download table itself

**Comment by amlcurran** [comment]
what do you need it for?

**Comment by xrigau** [comment]
secret

**Comment by ronocod** [comment]
:+1: LGTM

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ private DownloadContract() {
          * <P>Type: BIGINT</P>
          * <P>Owner can Read</P>
          */
-        public static final String COLUMN_LAST_MODIFICATION = "lastmod";
+        public static final String COLUMN_LAST_MODIFICATION = "last_modified_timestamp";
```
Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java]: API breaking change

**Comment by amlcurran** [comment]
:ship: in ten

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java (hunk 2 above)
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java]: why?

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java]: the public API is the constant, not the string itself

Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadContract.java]: I disagree

**Comment by friedger** [comment]
11 minutes are over

**Comment by ronocod** [comment]
:watch: :keycap_ten:

**Comment by xrigau** [comment]
for merge

**Comment by danybony** [comment]
LGTM :+1:

File: library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java (hunk 1 above)
Comment (ouchadam) [inline library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java]: I would have argued for a `last_status_modification` :tongue:

**Pull request merged**